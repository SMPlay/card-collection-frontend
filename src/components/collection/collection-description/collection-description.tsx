import React, { FC } from "react";
import makeStyles from "@material-ui/core/styles/makeStyles";

import { CollectionType } from "../../../types/CollectionType";
import { Card, CardMedia, Grid, Typography } from "@material-ui/core";

export interface CollectionDescriptionProps {
  description: CollectionType
}

const useStyles = makeStyles(
  {
    card: {
      width: 300
    }
  }
);

export const CollectionDescription: FC<CollectionDescriptionProps> = ({description}) => {
  const styles = useStyles();

  const {release, imageUrl, cardsCount, name} = description;
  return (
    <Grid container spacing={2}>
      <Grid item>
        <Card className={styles.card}>
          <CardMedia
            component="img"
            alt={name}
            height="450"
            image={imageUrl}
            title={name}
          />
        </Card>
      </Grid>


      <Grid item>
        <Typography variant="h5" component="h2">
          Название: <b>{name}</b>
        </Typography>
        <Typography variant="h5" component="h2">
          Количество карт в коллекции: <b>{cardsCount}</b>
        </Typography>
        <Typography variant="h5" component="h2">
          Дата выпуска: <b>{release}</b>
        </Typography>
      </Grid>
    </Grid>
  )
}