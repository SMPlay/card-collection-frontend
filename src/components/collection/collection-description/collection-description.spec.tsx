import React from "react";
import { create } from "react-test-renderer";

import { CollectionDescription, CollectionDescriptionProps } from "./collection-description";

describe("CollectionDescription component", () => {
  it("should render", () => {
    const props: CollectionDescriptionProps = {
      description: {
        name: "test",
        imageUrl: "test",
        id: "test",
        cardsCount: 0,
        release: 0,
      }
    };

    const tree = create(<CollectionDescription {...props}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
