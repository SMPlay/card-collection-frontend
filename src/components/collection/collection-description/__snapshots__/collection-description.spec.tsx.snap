// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionDescription component should render 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <img
        alt="test"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
        height="450"
        src="test"
        title="test"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Название: 
      <b>
        test
      </b>
    </h2>
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Количество карт в коллекции: 
      <b>
        0
      </b>
    </h2>
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Дата выпуска: 
      <b>
        0
      </b>
    </h2>
  </div>
</div>
`;
